server/server.go

```diff
@@ -42,6 +42,10 @@ type StartSettings struct {
 
 	// Server's TLS configuration.
 	TLSConfig *tls.Config
+
+	// HTTPMiddlewareFunc specifies middleware for HTTP messages received by the server.
+	// This function is optional to set.
+	HTTPMiddlewareFunc func(handlerFunc http.HandlerFunc) http.HandlerFunc
```

Comment:
I think it would be useful to add a comment explaining that the middleware func will be called for every message when HTTP transport is used and once per connection when WebSocket transport is used.

Reply:
good call, will update